src/inc/sstring.h

```diff
@@ -1006,6 +1006,9 @@ typedef InlineSString<512> StackSString;
 // be needed is small and it's preferable not to take up the stack space.
 typedef InlineSString<32>  SmallStackSString;
 
+// To be used specifically for path strings.
+typedef InlineSString<512> PathString;
```

Comment:
I would make this 256 (=existing MAX_PATH). Also, I would make the size much smaller in debug builds to make sure that we exercise the buffer allocation path.

```
#ifdef _DEBUG
// Make 
typedef InlineSString<32> PathString; 
#else
typedef InlineSString<256> PathString; 
#endif
```